tensorflow_privacy/privacy/membership_inference_attack/data_structures.py

```diff
@@ -444,6 +444,69 @@ def __str__(self):
     ])
 
 
+@dataclass
+class SingleRiskScoreResult:
+  """Results from computing privacy risk scores.
+  this part shows how to leverage privacy risk score to perform attacks with thresholding on risk score
+  """
+
+  # Data slice this result was calculated for.
+  slice_spec: SingleSliceSpec
+
+  train_risk_scores: np.ndarray 
+  
+  test_risk_scores: np.ndarray
+  
+  def attack_with_varied_thresholds(self, threshold_list):
+    """ For each threshold value, we count how many training and test samples with privacy risk scores larger than the threshold
+    and further compute precision and recall values.
+    We skip the threshold value if it is larger than every sample's privacy risk score.
+    """
+    precision_list = []
+    recall_list = []
+    meaningful_threshold_list = []
+    for threshold in threshold_list:
+        true_positive_normalized = np.sum(self.train_risk_scores>=threshold)/(len(self.train_risk_scores)+0.0)
+        false_positive_normalized = np.sum(self.test_risk_scores>=threshold)/(len(self.test_risk_scores)+0.0)
+        if true_positive_normalized+false_positive_normalized>0:
+          meaningful_threshold_list.append(threshold)
+          precision_list.append(true_positive_normalized/(true_positive_normalized+false_positive_normalized+0.0))
+          recall_list.append(true_positive_normalized)
+    return np.array(meaningful_threshold_list), np.array(precision_list), np.array(recall_list)
+  
+  def collect_results(self, threshold_list=np.array([1,0.9,0.8,0.7,0.6,0.5])):
```

Comment:
Using default arguments with mutable types is not allowed by Google Python Styleguide
https://google.github.io/styleguide/pyguide.html#212-default-argument-values

Please remove using default arguments here.

The reason is that it's error-prone (more details in style guide)

Reply:
removed the default arguments